tagstudio/tests/qt/test_thumb_renderer.py

```diff
@@ -0,0 +1,32 @@
+# Copyright (C) 2024 Travis Abendshien (CyanVoxel).
+# Licensed under the GPL-3.0 License.
+# Created for TagStudio: https://github.com/CyanVoxel/TagStudio
+
+import io
+from pathlib import Path
+
+from PIL import Image
+from src.qt.widgets.thumb_renderer import ThumbRenderer
+from syrupy.extensions.image import PNGImageSnapshotExtension
+
+
+def test_odt_preview(cwd, snapshot):
+    file_path: Path = cwd / "fixtures" / "sample.odt"
+    renderer = ThumbRenderer()
+    img: Image.Image = renderer._open_doc_thumb(file_path)
+
+    img_bytes = io.BytesIO()
+    img.save(img_bytes, format="PNG")
+    img_bytes.seek(0)
+    assert img_bytes.read() == snapshot(extension_class=PNGImageSnapshotExtension)
+
+
+def test_ods_preview(cwd, snapshot):
+    file_path: Path = cwd / "fixtures" / "sample.ods"
+    renderer = ThumbRenderer()
+    img: Image.Image = renderer._open_doc_thumb(file_path)
```

Comment:
these two tests (and the tests in #543 #540 #539 ) can be eventually DRY-ed by using `pytest.mark.parametrize`. It executes the test with values in the supplied list, which works great in case the only thing which changes is the fixture name (and thumbnailing method in case of the other PRs). So it can look like this:


```suggestion
@pytest.mark.parametrize(["fixture_file", "thumbnailer"], [
    ("sample.odt", ThumbRenderer._open_doc_thumb), # _open_doc_thumb needs to be changed to @classmethod
    ("sample.ods", ThumbRenderer._open_doc_thumb), # _open_doc_thumb needs to be changed to @classmethod
])
def test_document_preview(cwd, fixture_file, thumbnailer, snapshot):
    file_path: Path = cwd / "fixtures" / fixture_file
    img: Image.Image = thumbnailer(file_path)
```

so instead of two almost-identical tests, there will be just one, and the fixture file + thumbnailing method is passed inside via the parameters.

Reply:
> these two tests (and the tests in #543 #540 #539 ) can be eventually DRY-ed by using `pytest.mark.parametrize`. It executes the test with values in the supplied list, which works great in case the only thing which changes is the fixture name (and thumbnailing method in case of the other PRs). So it can look like this:
> 
> so instead of two almost-identical tests, there will be just one, and the fixture file + thumbnailing method is passed inside via the parameters.

I've tried merging the tests as shown here, including marking `open_doc_thumb` as a `@classmethod`, however it seems that the snapshots can not be found causing the tests to fail. Here's how I have the test set up:

```py
@pytest.mark.parametrize(
    ["fixture_file", "thumbnailer"],
    [
        (
            "sample.odt",
            ThumbRenderer._open_doc_thumb,
        ),
        (
            "sample.ods",
            ThumbRenderer._open_doc_thumb,
        ),
    ],
)
def test_document_preview(cwd, fixture_file, thumbnailer, snapshot):
    file_path: Path = cwd / "fixtures" / fixture_file
    img: Image.Image = thumbnailer(file_path)

    img_bytes = io.BytesIO()
    img.save(img_bytes, format="PNG")
    img_bytes.seek(0)
    assert img_bytes.read() == snapshot(extension_class=PNGImageSnapshotExtension)
```
And the error from each run (same output as if the original tests had missing snapshots):
```
E       AssertionError: assert [+ received] == [- snapshot]
E         Snapshot 'test_document_preview[sample.odt-_open_doc_thumb]' does not exist!
E         + b'\x89PNG\r\n\x1
```

Reply:
Since the test was renamed, it's looking for a fixture matching the new test name + parameters, which doesnt exist yet. You have to run pytest with `--snapshot-update` the first time when a new test is created to generate the fixtures.